File: 백준/Silver/2108. 통계학/통계학.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

        int n = Integer.parseInt(br.readLine());
        int[] numbers = new int[n];
        
        int sum = 0;
        Map<Integer, Integer> frequency = new HashMap<>();

        for (int i = 0; i < n; i++) {
            numbers[i] = Integer.parseInt(br.readLine());
            sum += numbers[i];
            frequency.put(numbers[i], frequency.getOrDefault(numbers[i], 0) + 1);
        }

        Arrays.sort(numbers);

        int mean = (int) Math.round((double) sum / n);
        
        int median = numbers[n / 2];
        
        int maxFrequency = Collections.max(frequency.values());
        List<Integer> modes = new ArrayList<>();
        
        for (Map.Entry<Integer, Integer> entry : frequency.entrySet()) {
            if (entry.getValue() == maxFrequency) {
                modes.add(entry.getKey());
            }
        }
        
        Collections.sort(modes);
        int mode = modes.size() > 1 ? modes.get(1) : modes.get(0);
        
        int range = numbers[n - 1] - numbers[0];
        
        bw.write(mean + "\n");
        bw.write(median + "\n");
        bw.write(mode + "\n");
        bw.write(range + "\n");

        bw.flush();
        bw.close();
        br.close();
    }
}

File: 백준/Silver/2108. 통계학/통계학.py
import sys
from collections import Counter

input = sys.stdin.readline

N = int(input().strip())
data = [int(input().strip()) for _ in range(N)]

mean = round(sum(data) / N)

sorted_data = sorted(data)
median = sorted_data[N // 2]

count = Counter(data)
max_frequency = max(count.values())
mode_candidates = [num for num, freq in count.items() if freq == max_frequency]
mode = min(mode_candidates) if len(mode_candidates) == 1 else sorted(mode_candidates)[1]

range_val = max(data) - min(data)

result = [mean, median, mode, range_val]
sys.stdout.write('\n'.join(map(str, result)) + '\n')